pkg/network/mycelium/config.go

```diff
@@ -0,0 +1,56 @@
+package mycelium
+
+import (
+	"context"
+	"crypto/ed25519"
+	"encoding/hex"
+
+	"github.com/decred/base58"
+	"github.com/pkg/errors"
+)
+
+type NodeConfig struct {
+	KeyFile    string
+	TunName    string
+	Peers      []string
+	PrivateKey ed25519.PrivateKey
+	PublicKey  string
+	NodeInfo   map[string]interface{}
```

Comment:
What is this and what is it used for?

We don't need to copy everything over from ygg. Ygg has a completely different configuration and even requires a config file with way more config than mycelium.

Please drop all attributes that are not needed.